test/e2e_node/resource_collector.go

```diff
@@ -387,6 +389,28 @@ func deleteBatchPod(f *framework.Framework, pods []*api.Pod) {
 	return
 }
 
+// checkPodDeleted checks whether a pod has been successfully deleted
+func checkPodDeleted(f *framework.Framework, podName string) error {
+	_, err := f.Client.Pods(f.Namespace.Name).Get(podName)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	return errors.New("Pod Not Deleted")
+}
+
+// cleanupPerfTest deletes all pods created by performance related test which may create a lot of pods.
+// The pods may cause namespace controller times out when it tries to cleanup at the end of test.
+// So we explicitly cleanup here.
+func cleanupPerfTest(f *framework.Framework, pods []*api.Pod) {
```

Comment:
I don't think this needs to be a test-specific cleanup function. Looks like this could just be a general helper function which deletes all the pods passed to the function.